golem/model/src/main/scala/cloud/golem/runtime/plan/AgentImplementationPlan.scala

```diff
@@ -0,0 +1,47 @@
+package cloud.golem.runtime.plan
+
+import cloud.golem.data.GolemSchema
+import cloud.golem.runtime.{AgentMetadata, MethodMetadata}
+
+import scala.concurrent.Future
+
+final case class AgentImplementationPlan[Instance](
+  metadata: AgentMetadata,
+  buildInstance: () => Instance,
+  methods: List[MethodPlan[Instance]]
+)
+
+/**
+ * Agent implementation plan where the agent instance depends on constructor
+ * input.
```

Comment:
I think, if possible, we should simplify this idea. If the constructors don't exist, it's just empty. Having to separately handle them using different types feels like unnecessary. 




In other words, `AgentImplementationPlan` should never be a separate type. It is already covered in `AgentImplementationPlanWithCtor`. And thereby rename `AgentImplementationPlanWithCtor[Instance, Ctor]` to just `AgentImplementationPlan[Instance, Ctor]`